git/object_scanner.go

```diff
@@ -52,55 +41,26 @@ type ObjectScanner struct {
 //
 // Otherwise, an `*ObjectScanner` is returned with no error.
 func NewObjectScanner() (*ObjectScanner, error) {
-	cmd := gitNoLFS("cat-file", "--batch")
-	stdout, err := cmd.StdoutPipe()
+	gitdir, err := GitCommonDir()
 	if err != nil {
-		return nil, errors.Wrap(err, "open stdout")
-	}
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		return nil, errors.Wrap(err, "open stdin")
+		return nil, err
 	}
 
-	stderr, err := cmd.StderrPipe()
+	gitobj, err := gitobj.FromFilesystem(filepath.Join(gitdir, "objects"), "")
 	if err != nil {
-		return nil, errors.Wrap(err, "open stderr")
-	}
-
-	closeFn := func() error {
-		if err := stdin.Close(); err != nil {
-			return err
-		}
-
-		msg, _ := ioutil.ReadAll(stderr)
-		if err = cmd.Wait(); err != nil {
-			return errors.Errorf("Error in git cat-file --batch: %v %v", err, string(msg))
-		}
-
-		return nil
-	}
-
-	tracerx.Printf("run_command: git cat-file --batch")
-	if err := cmd.Start(); err != nil {
 		return nil, err
 	}
 
 	return &ObjectScanner{
-		from: bufio.NewReaderSize(stdout, 16384),
-		to:   stdin,
-
-		closeFn: closeFn,
+		gitobj: gitobj,
```

Comment:
Does it make sense here to instead say:

```go
return NewObjectScannerFrom(gitobj), nil
```

instead? I think the two are synonyms right now, but potentially the body of the later would change and we'd minimize the diff later.

Reply:
Fixed.